javalib/src/main/scala/java/time/LocalTime.scala

```diff
@@ -0,0 +1,381 @@
+        new LocalTime(hour, value.toInt, second, nano)
+
+      case MINUTE_OF_DAY =>
+        require(value >= 0 && value <= MINUTES_IN_DAY, msg)
+        ofNanoOfDay(value * NANOS_IN_MINUTE + second.toLong * NANOS_IN_SECOND + nano)
+
+      case HOUR_OF_AMPM =>
+        require(value >= 0 && value <= 11, msg)
+        if (hour < 12) new LocalTime(value.toInt, minute, second, nano)
+        else new LocalTime(value.toInt + 12, minute, second, nano)
+
+      case CLOCK_HOUR_OF_AMPM =>
+        require(value >= 1 && value <= 12, msg)
+        `with`(HOUR_OF_AMPM, value % 12)
+
+      case HOUR_OF_DAY =>
+        require(value >= 0 && value <= 23, msg)
+        new LocalTime(value.toInt, minute, second, nano)
+
+      case CLOCK_HOUR_OF_DAY =>
+        require(value >= 1 && value <= 24, msg)
+        new LocalTime(value.toInt % 24, minute, second, nano)
+
+      case AMPM_OF_DAY =>
+        require(value >= 0 && value <= 1, msg)
+        new LocalTime(hour % 12 + value.toInt * 12, minute, second, nano)
+
+      case _: ChronoField =>
+        throw new UnsupportedTemporalTypeException(s"Field not supported: $field")
+
+      case _ => field.adjustInto(this, value)
+    }
+  }
+
+  def withHour(hour: Int): LocalTime =
+    new LocalTime(hour, minute, second, nano)
+
+  def withMinute(minute: Int): LocalTime =
+    new LocalTime(hour, minute, second, nano)
+
+  def withSecond(second: Int): LocalTime =
+    new LocalTime(hour, minute, second, nano)
+
+  def withNano(nanoOfSecond: Int): LocalTime =
+    new LocalTime(hour, minute, second, nanoOfSecond)
+
+  def truncatedTo(unit: TemporalUnit): LocalTime = {
+    if (unit.getDuration.getSeconds > SECONDS_IN_DAY) {
+      throw new UnsupportedTemporalTypeException("Unit too large")
+    } else if (NANOS_IN_DAY % unit.getDuration.toNanos != 0) {
+      val msg = "Unit must divide into a standard day without remainder"
+      throw new UnsupportedTemporalTypeException(msg)
+    } else {
+      val unitNanos = unit.getDuration.toNanos
+      ofNanoOfDay(totalNanos / unitNanos * unitNanos)
+    }
+  }
+
+  override def plus(amount: TemporalAmount): LocalTime =
+    amount.addTo(this).asInstanceOf[LocalTime]
+
+  def plus(amount: Long, unit: TemporalUnit): LocalTime = unit match {
+    case NANOS     => plusNanos(amount)
+    case MICROS    => plusNanos((amount % MICROS_IN_DAY) * NANOS_IN_MICRO)
+    case MILLIS    => plusNanos((amount % MILLIS_IN_DAY) * NANOS_IN_MILLI)
+    case SECONDS   => plusSeconds(amount)
+    case MINUTES   => plusMinutes(amount)
+    case HOURS     => plusHours(amount)
+    case HALF_DAYS => plusHours((amount % 2) * 12)
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+
+    case _ => unit.addTo(this, amount).asInstanceOf[LocalTime]
+  }
+
+  def plusHours(hours: Long): LocalTime = {
+    val offset = if (hours < 0) hours % 24 + 24 else hours % 24
+    new LocalTime((hour + offset.toInt) % 24, minute, second, nano)
```

Comment:
You could improve performance by doing operations on `Int` rather than longs:
```scala
val offset = (hours % 24).toInt + 24
new LocalTime((hour + offset) % 24, minute, second, nano)
```
In Scala.js `Int` operations are more performant than `Long` operations.